src/core/sync/mutex.d

```diff
@@ -163,7 +167,7 @@ class Mutex :
         lock_nothrow();
     }
 
-    // undocumented function for internal use
+    /// ditto
     final void lock_nothrow(this Q)() nothrow @trusted @nogc
```

Comment:
Since those methods (lock, unlock, tryLock) are virtual, we can't add attributes to them without breaking code. @MartinNowak attempted to enforce `nothrow` back then when 2.067 was to be released, but that broke `vibe.d` and had to be [reverted](https://github.com/dlang/druntime/pull/1188). You can read more details about why vibe.d needed to throw from the locking primitives in [this](https://github.com/rejectedsoftware/vibe.d/pull/972) thread (in short - in order to implement interruptable tasks with proper resource cleanup).

I'm not 100% sure if we should document those functions, but that's what we should actually try to advertise as it is more friendly to code that tries to apply attributes where possible. I'm also not fan of the names, but considering that they're already public (albeit undocumented) I think we're kind of stuck with them. The only alternative names that I can think of are acquire and release, though tryAcquire sound bad to me.

Perhaps we can add `lockExcept` / `unlockExcept` virtual functions and deprecate the throwing methods, but I prefer not to do that in this change.